### PR TITLE
better design for schedualers and disabling schedualers on dev mode (configurable)

### PR DESCRIPTION
**Pull request by eishay**
@andrewconner @yingjieMiao please review

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/common/actor/ActorSystemModule.scala [hunk 1]
```diff
@@ -43,20 +40,16 @@ case class DevActorSystemModule() extends ActorSystemModule {
...
-    (current.configuration.getBoolean("scheduler.enabled").map {
-      case false => SchedulingEnabled.Never
-      case true => SchedulingEnabled.Always
-    }).getOrElse(SchedulingEnabled.Never)
+  def globalSchedulingEnabled: SchedulingProperties = {
+    val enabledProp = current.configuration.getBoolean("scheduler.enabled").getOrElse(false)
+    new SchedulingProperties {
+      def enabled = enabledProp
+      def enabledOnlyForLeader = enabledProp
+    }
```
Comment (yingjieMiao) [inline kifi-backend/modules/common/app/com/keepit/common/actor/ActorSystemModule.scala:48]: these two functions always return the same value?

Comment (eishay) [inline kifi-backend/modules/common/app/com/keepit/common/actor/ActorSystemModule.scala:48]: which two?

**Comment by raymondng** [comment]
FYI -- confirmed with @eishay that plugins are not disabled in dev mode

File: kifi-backend/modules/common/app/com/keepit/common/actor/ActorSystemModule.scala (hunk 1 above)
Comment (yingjieMiao) [inline kifi-backend/modules/common/app/com/keepit/common/actor/ActorSystemModule.scala:48]: enabled and enabledOnlyForLeader. These two always take same value?

Comment (eishay) [inline kifi-backend/modules/common/app/com/keepit/common/actor/ActorSystemModule.scala:48]: true, in dev mode they have the same value which is the value from `application-dev.conf`

Comment (eishay) [inline kifi-backend/modules/common/app/com/keepit/common/actor/ActorSystemModule.scala:48]: note that in prod mode the implementation is diferant:

``` scala
def enabledOnlyForLeader: Boolean = enabled && serviceDiscovery.isLeader()
```

Comment (yingjieMiao) [inline kifi-backend/modules/common/app/com/keepit/common/actor/ActorSystemModule.scala:48]: thanks

**Pull request merged**